rocAL/rocAL/source/api/rocal_api_meta_data.cpp

```diff
@@ -231,8 +231,16 @@ ROCAL_API_CALL rocalGetImageLabels(RocalContext p_context, void* buf, unsigned i
     if(context->user_batch_size() != meta_data_batch_size)
         THROW("meta data batch size is wrong " + TOSTR(meta_data_batch_size) + " != "+ TOSTR(context->user_batch_size() ))
 
-    if (context->affinity == RocalAffinity::CPU)
-        memcpy(buf, meta_data.second->get_label_batch().data(), sizeof(int) * meta_data_batch_size);
+    if (context->affinity == RocalAffinity::CPU) {
+        if (!flags) {
+            memcpy(buf, meta_data.second->get_label_batch().data(), sizeof(int) * meta_data_batch_size);
```

Comment:
Can we not directly assign it to pointer?

Reply:
Memcpy needs to happen otherwise the labels maybe overwritten during the next load routine while training is still running